src/js/components/VendingMachineManageMenu/VendingMachineManageMenu.js

```diff
@@ -0,0 +1,132 @@
+import { SELECTOR } from '../../constants/selector.js';
+import { COINS, COIN_10, COIN_100, COIN_50, COIN_500 } from '../../constants/vendingMachineManageMenu.js';
+import { $ } from '../../utils/dom.js';
+import { InvalidValueError } from '../../utils/error.js';
+import { chargeStorage, coinsStorage } from '../../utils/storage.js';
+import { validateVendingMachineCharge } from '../../utils/validation.js';
+
+/* eslint-disable class-methods-use-this */
+export default class VendingMachineManageMenu {
+  #state = {
+    charge: 0,
+    coins: {},
+  };
+
+  get state() {
+    return this.#state;
+  }
+
+  set state(state) {
+    this.#state = state;
+  }
+
+  init() {
+    this.#state = this.#getInitialState();
+    this.#render();
+    this.#focusInput();
+    this.#bindEvents();
+  }
+
+  #getInitialState() {
+    return {
+      charge: chargeStorage.get('charge'),
+      coins: coinsStorage.get('coins'),
+    };
+  }
+
+  #focusInput() {
+    $(SELECTOR.VENDING_MACHINE_CHARGE_INPUT).focus();
+  }
+
+  #resetInput() {
+    $(SELECTOR.VENDING_MACHINE_CHARGE_INPUT).value = '';
+  }
+
+  #getCoins(charge, units) {
+    let newCharge = charge;
+    const coins = {};
+
+    units.forEach((unit) => {
+      coins[unit] = Math.floor(newCharge / unit);
+      // eslint-disable-next-line no-bitwise
+      newCharge %= unit;
+    });
```

Comment:
```suggestion
    const { coins } = units.reduce((prev, acc) => {
        const currentCoin = Math.floor(prev.charge / acc);
        const newCoins = { ...prev.coins, [acc]: currentCoin };
        const remainCharge = prev.charge - currentCoin * acc;
        
        return { coins: newCoins, charge: remainCharge }
    }, { coins: {}, charge })
```
성능적으로 더 좋지는 않겠지만 이렇게 작성하면 lint룰도 회피할 수 있고 동작도 예측하기 수월할거라고 생각해요 👀

Reply:
`이 부분 로직을 어떻게 작성하는게 효과적일까?`라고 고민을 했었는데 확실히 동작을 예측하기에 좋아진 것 같습니다.
배열을 `forEach`를 돌릴거라면, `reduce`를 먼저 고민해봤어도 좋았을 거라는 생각도 드네요.

`remainCharge`를 구하는 로직이 직관적인게 마음에 듭니다. 제안 감사합니다!